### PR TITLE
virtiofs: update to use virtio-queue 0.1.0

### PR DESCRIPTION
**Pull request by bergwolf**
So that we no longer rely on a non-release git commit.

### CODE REVIEW

File: src/transport/virtiofs/mod.rs [hunk 1]
```diff
@@ -105,10 +105,11 @@ impl std::error::Error for Error {}
 
 impl<'a> Reader<'a> {
     /// Construct a new Reader wrapper over `desc_chain`.
-    pub fn new(
-        mem: &'a GuestMemoryMmap,
-        desc_chain: DescriptorChain<GuestMemoryAtomic<GuestMemoryMmap>>,
-    ) -> Result<Reader<'a>> {
+    pub fn new<M>(mem: &'a GuestMemoryMmap, desc_chain: DescriptorChain<M>) -> Result<Reader<'a>>
```
Comment (jiangliu) [inline src/transport/virtiofs/mod.rs:108]: Should we use "mem: M" instead of "mem: &'a GuestMemoryMmap"?

**Comment by jiangliu** [comment]
It would be great to publish v0.2 once this MR get merged.

File: src/transport/virtiofs/mod.rs (hunk 1 above)
Comment (bergwolf) [reply src/transport/virtiofs/mod.rs:108]: Do you mean adding `M: Deref + GuestMemory`? vhost-user-backend requires `GuestMemoryAtomic<GuestMemoryMmap>` but I don't see how to convert it back to `GuestMemory`. Looks like vm-memory is missing some abstractions, e.g.:

```
the trait `vm_memory::GuestMemory` is not implemented for `Arc<GuestMemoryMmap>`
```

Comment (jiangliu) [reply src/transport/virtiofs/mod.rs:108]: Does this work?
```
pub fn new<M>(mem: M, desc_chain: DescriptorChain<M>) -> Result<Reader<'a>> 
    where
        M: Deref,
        M::Target: GuestMemory + Sized,
    {
```

Comment (bergwolf) [reply src/transport/virtiofs/mod.rs:108]: In that case, I'm getting:

> 
   --> src/transport/virtiofs/mod.rs:136:14
    |
136 |             .collect::<Result<VecDeque<VolatileSlice<'a>>>>()?;
    |              ^^^^^^^ value of type `VecDeque<VolatileSlice<'a>>` cannot be built from `std::iter::Iterator<Item=VolatileSlice<'_, <<<<M as std::ops::Deref>::Target as vm_memory::GuestMemory>::R as GuestMemoryRegion>::B as WithBitmapSlice<'_>>::S>>`

Comment (jiangliu) [reply src/transport/virtiofs/mod.rs:108]: then keep it as is, it will be solve by another way.

**Review by jiangliu — APPROVED** [review]

**Pull request merged**